### PR TITLE
Issue 51316: disable update of assay run/result sample lineage

### PR DESCRIPTION
**Pull request by labkey-nicka**
#### Rationale
This disables `AssayProvider.updatePropertyLineage()` processing in 24.9 and we'll revisit the implementation approach in a future release cycle. See [Issue 51316](https://www.labkey.org/home/Developer/issues/issues-details.view?issueId=51316).

#### Related Pull Requests
- #5755

#### Changes
- ~~In addition to the `row` data pass in the `oldRow` data to `AssayProvider.updatePropertyLineage()` when updating assay run/result rows.~~
- ~~Inspect the old row value for the given lookup and only remove a sample as a material input if it was the original value.~~
- Disable `AssayProvider.updatePropertyLineage()` processing and `@Ignore` associated regression test coverage.

### CODE REVIEW

**Comment by labkey-nicka** [comment]
I've kicked off a run against the LKSM suite for 24.9 in CI.

File: api/src/org/labkey/api/assay/AbstractAssayProvider.java [hunk 1]
```diff
@@ -2147,14 +2153,28 @@ public void updatePropertyLineage(
...
+                            // Issue 51316: Resolve the original material input rowId for this column
+                            Integer originalRowId = null;
+                            if (oldValue instanceof Integer _originalRowId)
+                                originalRowId = _originalRowId;
+                            else if (oldValue != null)
+                            {
+                                ExpMaterial originalMaterial = ExperimentService.get().findExpMaterial(container, user, oldValue, sampleType, cache, materialsCache);
+                                if (originalMaterial != null)
+                                    originalRowId = originalMaterial.getRowId();
+                            }
```
Comment (cnathe) [inline api/src/org/labkey/api/assay/AbstractAssayProvider.java:2176]: this currently would happened each iteration through the run material inputs. could this instead be moved above the for loop so that it only has to happen once?

**Review by cnathe — CHANGES_REQUESTED** [review]

**Comment by labkey-nicka** [comment]
@cnathe Alright, I've disabled the update functionality, updated the associated regression test to cover further edge cases, and disabled the associated regression test.

File: api/src/org/labkey/api/assay/AbstractAssayProvider.java (hunk 1 above)
Comment (labkey-nicka) [reply api/src/org/labkey/api/assay/AbstractAssayProvider.java:2176]: As discussed, we're looking to take a different approach to this problem so this implementation is effectively defunct.

**Review by cnathe — APPROVED** [review]
